creator/ingest_runs/tasks/validation_run.py

```diff
@@ -28,51 +32,108 @@
+    # file has already been mapped to the template keys
+    columns_to_keys = {}
+    for tv in template_versions:
+        for field in tv.field_definitions["fields"]:
+            key = field.get("key")
+            if key:
+                columns_to_keys[field["label"]] = key
+                columns_to_keys[key] = key
+
+    # Check for duplicate mappings (multiple source columns map to the same
+    # template key) and log them since this shouldn't really happen but if
+    # it does, could mean inconsistent data in the source file
+    reverse_mapping = defaultdict(set)
+    for src, key in columns_to_keys.items():
+        reverse_mapping[key].add(src)
+    duplicates = {
+        key: src_cols
+        for key, src_cols in reverse_mapping.items()
+        if len(src_cols) > 2
+    }
+    if duplicates:
         logger.warning(
-            f"⚠️  Could not find an extract config for file version {version} "
-            f"with file type: {version.root_file.file_type}. This file will "
-            "not be included in validation."
+            "Found multiple columns in source file(s) that map to the same "
+            f"template key:\n{pformat(duplicates)}"
         )
-        raise ExtractConfigFileDoesNotExist
+
+    return columns_to_keys
+
+
+def map_column(in_col: str, mapper: Dict[str, str]) -> str:
+    """
+    Map a column in a source file to a template key
+
+    The mapper dict contains a mapping of template columns to template
+    keys
+
+    1. (Not Implemented Yet) Try to fuzzy match the src file column to a
+    template column
+    2. Map result column from 1 to a template key
+    """
+    return mapper.get(in_col.strip())
+
+
+def clean_and_map(
+    version: Version, mapper: Dict[str, str]
+) -> pandas.DataFrame:
+    """
+    Extract tabular data from a file version into a pandas.DataFrame. Then
+    map the DataFrame columns to the file's template keys. This produces a
+    DataFrame with standard columns that can be recognized by the validator
+    """
+    logger.info(
+        f"Attempting to map file {version_display(version)} to template keys"
+    )
     # Extract file content into a DataFrame
     try:
         df = pandas.DataFrame(extract_data(version))
     except Exception as e:
         er_msg = (
-            f"Validation failed! Error in parsing {version} content into "
-            f"a DataFrame."
+            f"Error in parsing {version_display(version)} "
+            "content into a DataFrame."
         )
-        raise type(e)(er_msg).with_traceback(sys.exc_info()[2])
+        raise ExtractDataError from e
 
-    logger.info(
-        f"Applied extract config {os.path.split(ec)[-1]} for file "
-        f"{version.root_file.file_type}, {version.kf_id}"
-    )
+    # Try to map as many of the input columns to template keys
+    mapped_cols = {c: map_column(c, mapper) for c in df.columns}
+    mapped_df = df.rename(columns=mapped_cols, errors="ignore")
```

Comment:
I think you can condense the two above lines to:
```
mapped_df = df.rename(columns=map_column, errors="ignore")
```

Reply:
The `map_column` function still needs the `mapper` input and I think `rename` only lets you pass in a func